doc/index.md

```diff
@@ -68,31 +68,53 @@ To run the testsuite, you'll also need:
-LXD consists of two binaries, a client called `lxc` and a server called `lxd`.
-These live in the source tree in the `lxc/` and `lxd/` dirs, respectively.
-To get the code, set up your go environment:
+### From Source: Building the latest version
+To build the latest version of the code, perform the following steps:
 
 ```bash
 mkdir -p ~/go
 export GOPATH=~/go
+go get -d -v github.com/lxc/lxd/lxd
+cd $GOPATH/src/github.com/lxc/lxd
+make deps
+export CGO_CFLAGS="${CGO_CFLAGS} -I${GOPATH}/deps/sqlite/ -I${GOPATH}/deps/dqlite/include/ -I${GOPATH}/deps/raft/include/ -I${GOPATH}/deps/libco/"
+export CGO_LDFLAGS="${CGO_LDFLAGS} -L${GOPATH}/deps/sqlite/.libs/ -L${GOPATH}/deps/dqlite/.libs/ -L${GOPATH}/deps/raft/.libs -L${GOPATH}/deps/libco/"
+export LD_LIBRARY_PATH="${GOPATH}/deps/sqlite/.libs/:${GOPATH}/deps/dqlite/.libs/:${GOPATH}/deps/raft/.libs:${GOPATH}/deps/libco/:${LD_LIBRARY_PATH}"
+make
 ```
 
-And then download it as usual:
+The build process will use `go get` and `git clone` to grab all necessary dependencies needed for building.
+
+### From Source: Building a Release
+
+To build an official release of LXD, download and extract a release tarball, and then set up GOPATH to point to the
+`_dist` directory inside it, which is configured to be used as a GOPATH and contains snapshots of all necessary sources. LXD
+will then build using these snapshots rather than grabbing 'live' sources using `go get` and `git clone`:
 
 ```bash
-go get -d -v github.com/lxc/lxd/lxd
-cd $GOPATH/src/github.com/lxc/lxd
+cd lxd-3.18
+export GOPATH=$(pwd)/_dist
 make deps
 export CGO_CFLAGS="${CGO_CFLAGS} -I${GOPATH}/deps/sqlite/ -I${GOPATH}/deps/dqlite/include/ -I${GOPATH}/deps/raft/include/ -I${GOPATH}/deps/libco/"
 export CGO_LDFLAGS="${CGO_LDFLAGS} -L${GOPATH}/deps/sqlite/.libs/ -L${GOPATH}/deps/dqlite/.libs/ -L${GOPATH}/deps/raft/.libs -L${GOPATH}/deps/libco/"
 export LD_LIBRARY_PATH="${GOPATH}/deps/sqlite/.libs/:${GOPATH}/deps/dqlite/.libs/:${GOPATH}/deps/raft/.libs:${GOPATH}/deps/libco/:${LD_LIBRARY_PATH}"
 make
 ```
 
-...which will give you two binaries in `$GOPATH/bin`, `lxd` the daemon binary,
-and `lxc` a command line client to that daemon.
+### From Source: Installing
+
+When the build process completes,`$GOPATH/bin` will contain the `lxd` daemon, the `lxc` binary, as well as other tools.
+Binaries can be copied to appropriate locations in your path. The `$GOPATH/deps` directory will contain versions of
+requisite libraries that will also need to be installed in order for LXD to function. These libraries can be installed
+using the following commands:
+
+```bash
+cd "${GOPATH}/deps/sqlite" && make DESTDIR=/ libdir=/usr/lib install
+cd "${GOPATH}/deps/raft" && make DESTDIR=/ libdir=/usr/lib install
+cd "${GOPATH}/deps/libco" && make DESTDIR=/ LIBDIR=/usr/lib install
+cd "${GOPATH}/deps/dqlite" && make DESTDIR=/ libdir=/usr/lib install
+```
+`DESTDIR` and/or `libdir`/`LIBDIR` can be adjusted as necessary to install the libraries to a suitable library path.
```

Comment:
Might be worth mentioning in this section or in a separate one, that installing isn't necessary or even useful during development. Instead in such cases you just want to ensure that you have the variables from `make deps` set (usually through bashrc) and that you have `~/go/bin` in your PATH, at which point you can just run any of the binaries directly.